tests/ReadPropertiesTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace DigipolisGent\Tests\Robo\Task\General;
+
+use League\Container\ContainerAwareInterface;
+use League\Container\ContainerAwareTrait;
+use Robo\Contract\ConfigAwareInterface;
+use Robo\Common\CommandArguments;
+use Robo\Robo;
+use Robo\TaskAccessor;
+use Symfony\Component\Console\Output\NullOutput;
+
+class ReadPropertiesTest extends \PHPUnit_Framework_TestCase implements ContainerAwareInterface, ConfigAwareInterface
+{
+
+    use \DigipolisGent\Robo\Task\General\loadTasks;
+    use TaskAccessor;
+    use ContainerAwareTrait;
+    use CommandArguments;
+    use \Robo\Task\Base\loadTasks;
+    use \Robo\Common\ConfigAwareTrait;
+
+    /**
+     * Set up the Robo container so that we can create tasks in our tests.
+     */
+    public function setUp()
+    {
+        $container = Robo::createDefaultContainer(null, new NullOutput());
+        $this->setContainer($container);
+        $this->setConfig(Robo::config());
+    }
+
+
+    protected function getRandomString($length = 5)
+    {
+        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+        $charactersLength = strlen($characters);
+        $randomString = '';
+        for ($i = 0; $i < $length; $i++) {
+            $randomString .= $characters[rand(0, $charactersLength - 1)];
+        }
+        return $randomString;
+    }
+
+    /**
+     * Scaffold the collection builder.
+     *
+     * @return \Robo\Collection\CollectionBuilder
+     *   The collection builder.
+     */
+    public function collectionBuilder()
+    {
+        $emptyRobofile = new \Robo\Tasks();
+
+        return $this->getContainer()
+            ->get('collectionBuilder', [$emptyRobofile]);
+    }
+
+    public function testRun() {
+        // Test with root config and overrides.
+        $this->getConfig()->set('digipolis.root.project', __DIR__ . '/root');
+        $result = $this->taskReadProperties(__DIR__ . '/web')
```

Comment:
Do you think the properties files will usually be put in the web folder? That would mean you'd need a Robofile.php in the web folder too according to the current implementation. I'd think there'd be a organisation package with a default.properties.yml file for the default properties and a properties.yml file in the project root or the subfolder of the root.